Tests/test_file_ppm.py

```diff
@@ -240,7 +240,7 @@ def test_header_token_too_long(tmp_path):
 def test_truncated_file(tmp_path):
     # Test EOF in header
     path = str(tmp_path / "temp.pgm")
-    with open(path, "w") as f:
+    with open(path, "w", encoding="utf-8") as f:
         f.write("P6")
 
     with pytest.raises(ValueError) as e:
```

Comment:
ppm/pgm files are either [ASCII or binary](https://en.wikipedia.org/wiki/Netpbm#Description). Since we're writing text here we should probably use `"ascii"`.

Reply:
Since this is P6, it is binary.

I've created PR #6677 to change this to write in binary format.